crates/controller/src/mouse.rs

```diff
@@ -22,7 +23,9 @@ impl Plugin for MousePlugin {
                     .with_system(
                         update_position
                             .run_in_state(GameState::Playing)
-                            .label(MouseLabels::Position),
+                            .label(MouseLabels::Position)
+                            .after(HUDInteraction::Click)
```

Comment:
Why is this needed? Is it because you rely on mouse input clearing (`mouse.clear();`) inside the system with this label?

I think that this is easy to miss subtlety (a mine which will cause bugs in the future). I think that we should come up with a more robust input handling mechanism.

Also, what happens if the mouse is clicked while over HUD but no button?

Reply:
My idea was almost what you described. I wanted to create the set of steps by which the click would be processed. But as you correctly pointed out it would not cover the clicks on all visible HUD. Will need to rethink this..